### PR TITLE
Initial commit week3

### PR DESCRIPTION
**Pull request by nielstj**

### CODE REVIEW

**Review by chshersh — APPROVED** [review]
This looks nice to me 👍

File: week3/golf.hs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module Golf where
+
+import Data.List
+  
+-- Exercise 1
+skips :: [a] -> [[a]]
+skips [] = []
+skips a@(_:xs) = a : skips xs
```
Comment (JolandaNava) [inline week3/golf.hs]: For `[a,b,c,d]` your `skips` returns: `[[a,b,c,d], [b,c,d], [c,d], [d]]`, but it should return `[[a,b,c,d], [b,d], [c], [d]]`

File: week3/golf.hs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+-- Exercise 2
+bundle :: Int -> [Integer] -> [[Integer]]
+bundle _ [] = []
+bundle n a@(_:xs) = filter (\x -> length x == n) 
+                    ((take n a) : bundle n xs)
+
+localMaxima :: [Integer] -> [Integer]
+localMaxima = map (\(_:b:_) -> b) 
+              . filter (\(a:b:c:_) -> (b > a) && (b > c)) 
+              . bundle 3
```
Comment (JolandaNava) [inline week3/golf.hs]: This is just a suggestion if you want to try and make this into a single function:
you are using `bundle` that takes `n` as a variable, but `n` never actually changes, it's just `3`, this is a hint that you might be able to encode the same logic using pattern-matching on the input instead.

For instance `localMaxima (x:xs@(y:z:_)) = ...` matches only lists with at least 3 elements. But you might have to use recursion to get this to work without an accumulator.

File: week3/golf.hs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+base :: String
+base = replicate 10 '=' ++ "\n0123456789\n"
+
+listToString :: [Integer] -> String
+listToString xs = map (\x -> case elem x xs of
+                                  True  -> '*'
+                                  False -> ' ') [0..9] ++ "\n\n"
+
+histogram :: [Integer] -> String
+histogram = (++ base) . concat . map listToString . reverse . transpose . group .sort
```
Comment (JolandaNava) [inline week3/golf.hs]: nice!

File: week3/golf.hs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+              . bundle 3
+
+-- Exercise 3
+base :: String
+base = replicate 10 '=' ++ "\n0123456789\n"
+
+listToString :: [Integer] -> String
+listToString xs = map (\x -> case elem x xs of
+                                  True  -> '*'
+                                  False -> ' ') [0..9] ++ "\n\n"
```
Comment (JolandaNava) [inline week3/golf.hs]: `case elem x xs of ...`  is the function [bool](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Bool.html#v:bool)

File: week3/golf.hs (hunk 1 above)
Comment (nielstj) [reply week3/golf.hs]: Fixed

File: week3/golf.hs (hunk 2 above)
Comment (nielstj) [reply week3/golf.hs]: Refactored with pattern match

File: week3/golf.hs (hunk 3 above)
Comment (nielstj) [reply week3/golf.hs]: Thank you!

File: week3/golf.hs (hunk 4 above)
Comment (nielstj) [reply week3/golf.hs]: Refactored with bool function

**Pull request merged**